code/modules/mob/living/silicon/robot/drone/drone.dm

```diff
@@ -297,18 +292,16 @@
 	to_chat(src, "<b>Don't invade their worksites, don't steal their resources, don't tell them about the changeling in the toilets.</b>")
 	to_chat(src, "<b>Make sure crew members do not notice you.</b>.")
 
-/*
-	sprite["Default"] = "repairbot"
-	sprite["Mk2 Mousedrone"] = "mk2"
-	sprite["Mk3 Monkeydrone"] = "mk3"
+	//now that the drone has a player, let's pick a sprite.
+	var/drone_icons[0]	//these are our sprite choices
+	drone_icons["Standard"] = "repairbot"
+	drone_icons["Pointy"] = "drone_scout"
+	drone_icons["Prismatic"] = "drone_gem"
+	drone_icons["Clockwork"] = "drone_clock"
 	var/icontype
-	icontype = input(player,"Pick an icon") in sprite
-	icon_state = sprite[icontype]
-	updateicon()
-
-	choose_icon(6,sprite)
-*/
-
+	icontype = input(player,"Choose a chassis design.") in drone_icons
+	icon_state = drone_icons[icontype]
```

Comment:
What if a player selects 'cancel' or closes the window?

Reply:
icon_state is set to 'repairbot' inherently, so cancelling the selection will just leave them with the standard drone sprite